Add preliminary support for Matrix

mytbk opened the pull request.

mytbk committed:
Add preliminary support for Matrix

bigeagle commented:
wow, cooool!
Hmmm, have you tested it?

bigeagle commented on README.md:
```diff
@@ -18,6 +18,7 @@ big party again.
 
 - IRC
 - XMPP
+- [Matrix](https://matrix.org) (depends on [matrix-python-sdk](https://github.com/matrix-org/matrix-python-sdk))
```
Please add this dependency to `requirements.txt`

mytbk replied:
done

bigeagle commented on fishroom/matrix.py:
```diff
@@ -0,0 +1,158 @@
+        except MatrixRequestError as e:
+            print(e)
+            if e.code == 403:
+                print("Bad username or password.")
```
use `logger.info` instead :smiley_cat:

mytbk replied:
done

mytbk commented:
The test() in matrix.py has not been tested yet. I've tested it with a Matrix room together with XMPP and IRC.

Multiple bindings has been tested with:

```
    'bindings': {
        "archlinux-cn": {
            "xmpp": 'archlinux@conference.cnjabber.net',
            "matrix": '#archlinux:matrixim.cc'
        },

        "linuxmasterrace": {
            "matrix": '#linuxmasterrace:matrix.org',
            "xmpp": 'linuxmasterrace@conference.cnjabber.net'
        }
    }
```


Only m.text msgtype in m.room.message event (https://matrix.org/docs/spec/client_server/r0.2.0.html#m-room-message-msgtypes) is supported.

Nick is not supported.

bigeagle commented:
OK. I've commented on the code, you can fix the little problems that I've pointed out. I'll merge this as soon as you push the updates.

mytbk committed:
Matrix: nick support

- set nick after logged in when nick is set in config.py
- cache and show Matrix user's display name

mytbk commented:
I've added nick support in another commit.

bigeagle: I'm sorry, but I can't see where your comment is.

bigeagle commented:
I'm sorry I forgot to submit my comments. :cry:

mytbk committed:
Add matrix-client to requirements.txt

mytbk committed:
Use logger for better logging.
